web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -205,9 +205,24 @@ export class LayerDialogComponent implements OnDestroy {
       });
   }
 
-  onClose() {
-    this.dialogRef.close();
-    return this.navigationService.selectProject(this.projectId!);
+  onCancel(): void | undefined {
+    if (!this.hasUnsavedChanges()) {
+      this.onClose();
+      return;
+    }
+    this.dialogService
+      .openConfirmationDialog(
+        'Warning',
+        'You have made unsaved changes to the layer. Would you like to save them?'
```

Comment:
"Unsaved changes to this layer will be lost. Are you sure?"

And set buttons to:
"CANCEL" and "DISCARD" as per https://material.io/components/dialogs#usage

Be sure to update the conditions in the subscribe callback below as well.